loleaflet/src/core/Socket.js

```diff
@@ -780,9 +780,16 @@ L.Socket = L.Class.extend({
 				var docUrl = url.split('?')[0];
 				this._map.options.doc = docUrl;
 				this._map.options.wopiSrc = encodeURIComponent(docUrl);
+
+				// if this is save-as, we need to load the document with edit permission
+				// otherwise the user has to close the doc then re-open it again
+				// in order to be able to edit.
+				if (textMsg.startsWith('saveas:'))
+					this._map.options.permission = 'edit';
```

Comment:
I would expect the fix to be inside WSD, not the client. Can't WSD simply re-open the document as writable after the save-as, and that automatically sends the permission as 'edit'. Then we don't have to do anything in the JS code.

Reply:
AFAIK, we do save-as on the integration, then  the integration sends the new wopi url which is passed to js inside saveas: cmd. Then the js is just sets the new WOPI url then send load again to LOOLWSD with the new url. So loadDocument command has the parameter readonly inside the js.

Reply:
Makes sense. So this is the "load" part of the equation.